WordPress/src/main/java/org/wordpress/android/viewmodel/SingleEventObservable.kt

```diff
@@ -0,0 +1,32 @@
+package org.wordpress.android.viewmodel
+
+import android.arch.lifecycle.LifecycleOwner
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.Observer
+
+/**
+ * A lifecycle-aware observable that sends only new updates after subscription, used for events like
+ * navigation and SnackBar messages.
+ *
+ *
+ * This avoids a common problem with events: on configuration change (like rotation) an update
+ * can be emitted if the observer is active.
+ *
+ *
+ * Note that only one observer can be subscribed.
+ */
+class SingleEventObservable<T>(private val sourceLiveData: LiveData<T>) {
+    private var lastEvent: T? = null
+
+    fun observe(owner: LifecycleOwner, observer: Observer<T>) {
+        if (sourceLiveData.hasObservers()) {
+            throw IllegalStateException("SingleEventObservable can be observed only by a single observer.")
+        }
+        sourceLiveData.observe(owner, Observer {
+            if (it !== lastEvent) {
```

Comment:
I am not sure if instance comparison is exactly what we want here. I know we don't want a value comparison, because it's totally valid to, for example, show the same toast message multiple times such as after a pull to refresh. I am skeptical about the instance creation as well, because it looks like same event can be triggered multiple times as different instances and we end up having to handle it when we don't want to. Please check out main review comment where I'll share more details about my testing experience which is related to this.

One ugly alternative is to take an equals function in the constructor so we can let the view models decide how each event should be compared. For this PR, it'll probably mean that we'll compare the step or maybe step and state, but for a toast it might be an instance comparison.

Not being able to follow up the pending approach feels really bad, because ideally, it should work just like `SingleLiveEvent` without view model having to provide more logic. Let's maybe discuss this further on Monday after we can think about it a bit further.